extensions/shared/blocks/cover/cover-media-placeholder.js

```diff
@@ -0,0 +1,88 @@
+
+/**
+ * WordPress dependencies
+ */
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { useBlockEditContext } from '@wordpress/block-editor';
+import { useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import UpgradeNudge from "../../components/upgrade-nudge";
+import { videoFileExtensions } from './utils';
+import { isSimpleSite } from "../../site-type-utils";
+import getJetpackExtensionAvailability from "../../get-jetpack-extension-availability";
+
+/**
+ * Module Constants
+ */
+const ALLOWED_MEDIA_TYPES = [ 'image', 'video' ];
+
+/**
+ * Nudge shows when the user tries to upload a video file.
+ * Unlike the core/video block, handled/extended by the videopress block,
+ * the nudge is not shown permanently.
+ * It's handled by the MediaPlaceholder component.
+ * For this reason, we can't wrap the edit setting
+ * with the wrapPaidBlock() HOC, as the videopress does.
+ *
+ * @param {object}  props - Information about the user.
+ * @param {string}  props.name - Show the Nudge component.
+ * @param {boolean} props.show - Show the Nudge component.
+ * @returns {*} Nudge component or Null.
+ */
+const JetpackCoverUpgradeNudge = ( { name, show } ) =>
+	show
+		? <UpgradeNudge
+			plan="value_bundle"
+			blockName={ name }
+			title={ {
+				knownPlan: __( 'To use a video in this block, upgrade to %(planName)s.', 'jetpack' ),
+				unknownPlan: __( 'To use a video in this block, upgrade to a paid plan.', 'jetpack' ),
+			} }
+			subtitle={ false }
+		/>
+		: null;
+
+export default createHigherOrderComponent(
+	CoreMediaPlaceholder => props => {
+		const { name } = useBlockEditContext();
+		const { unavailableReason } = getJetpackExtensionAvailability( 'videopress' );
+		if (
+			( ! name || name !== 'core/cover' ) || // extend only for cover block
+			! isSimpleSite() || // only for Simple sites
+			! [ 'missing_plan', 'unknown' ].includes( unavailableReason )
+		) {
+			return <CoreMediaPlaceholder { ...props } />;
+		}
+
+		const { onError } = props;
+		const [ error, setError ] = useState( false );
```

Comment:
I'm not 100% that this willl obey [the Rules of Hooks](https://reactjs.org/docs/hooks-rules.html) because it's called conditionally depending on the result of the `if` above.

Reply:
I can move the state above.